numpyro/svi.py

```diff
@@ -177,15 +173,8 @@ def elbo(param_map, model, guide, model_args, guide_args, kwargs, constrain_fn,
     """
     param_map = constrain_fn(param_map)
     guide_log_density, guide_trace = log_density(guide, guide_args, kwargs, param_map)
-    if is_autoguide:
-        # in autoguide, a site's value holds intermediate value
-        for name, site in guide_trace.items():
-            if site['type'] == 'sample':
-                param_map[name] = site['value']
-    else:
-        # NB: we only want to substitute params not available in guide_trace
-        param_map = {k: v for k, v in param_map.items() if k not in guide_trace}
-        model = replay(model, guide_trace)
+    param_map = {k: v for k, v in param_map.items() if k not in guide_trace}
+    model = replay(model, guide_trace)
```

Comment:
@neerajprad I missed this issue: in SVI (not autoguide), we don't want to replay intermediate values. We only want to replay intermediate values in autoguide. So I guess we need to revert to the old logic too. Sorry for missing this!

Reply:
Sorry, that was my bad. This feature of supporting dynamic constraints has added more complexity to the code base than I had not anticipated earlier. I'll revert this in another PR.